### PR TITLE
MM-35073 Detox/E2E: Add e2e for remaining eligible mobile smoke tests

### PR DESCRIPTION
**Pull request by josephbaylon**
#### Summary
- Added tests for MM-T3200, MM-T3202, MM-T3203, MM-T3204, MM-T3205, MM-T3206, MM-T3207, MM-T3208, MM-T3209 in
    - `detox/e2e/test/smoke_test/direct_messages.e2e.js`
    - `detox/e2e/test/smoke_test/private_channels.e2e.js`
    - `detox/e2e/test/smoke_test/public_channels.e2e.js`
- Updated helper functions

#### Ticket Link
JIRA tickets:
- https://mattermost.atlassian.net/browse/MM-35073

Zephyr cases:
- [MM-T3200](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T3200)
- [MM-T3202](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T3202)
- [MM-T3203](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T3203)
- [MM-T3204](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T3204)
- [MM-T3205](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T3205)
- [MM-T3206](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T3206)
- [MM-T3207](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T3207)
- [MM-T3208](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T3208)
- [MM-T3209](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T3209)

Screen Shot 2021-04-26 at 4 06 57 PM
Screen Shot 2021-04-26 at 4 03 42 PM
Screen Shot 2021-04-26 at 4 04 09 PM

### CODE REVIEW

**Review by saturninoabril — APPROVED** [review]
Thanks Joseph! Tested and passed.

File: package.json [hunk 1]
```diff
@@ -162,6 +162,7 @@
     "android": "react-native run-android",
     "ios": "react-native run-ios",
     "start": "react-native start",
+    "start:reset-cache": "react-native start -- --reset-cache",
```
Comment (enahum) [inline package.json:165]: 👍

**Review by enahum — APPROVED** [review]

**Pull request merged**